src/sentry/integrations/bitbucket/repository.py

```diff
@@ -0,0 +1,163 @@
+from __future__ import absolute_import
+
+import six
+
+from uuid import uuid4
+
+from sentry.app import locks
+from sentry.models import OrganizationOption
+from sentry.plugins import providers
+from sentry.models import Integration
+from sentry.utils.http import absolute_uri
+
+from sentry_plugins.exceptions import ApiError
+
+from .webhook import parse_raw_user_email, parse_raw_user_name
+
+
+class BitbucketRepositoryProvider(providers.IntegrationRepositoryProvider):
+    name = 'Bitbucket v2'
+
+    def raise_error(self, error, identity):
+        # TODO(LB): This used to do a lot more. Not sure it's important to handle atm
+        raise error
+
+    def get_client(self, integration_id):
+        if integration_id is None:
+            raise ValueError('Bitbucket version 2 requires an integration id.')
+
+        try:
+            integration_model = Integration.objects.get(id=integration_id)
+        except Integration.DoesNotExistError as error:
+            self.raise_error(error)
+
+        return integration_model.get_installation().get_client()
+
+    def get_config(self, organization):
+        choices = []
+        for i in Integration.objects.filter(organizations=organization, provider='bitbucket'):
+            choices.append((i.id, i.name))
+
+        if not choices:
+            choices = [('', '')]
+        return [
+            {
+                'name': 'integration_id',
+                'label': 'Bitbucket Integration',
+                'type': 'choice',
+                'choices': choices,
+                'initial': choices[0][0],
+                'help': 'Select which Bitbucket integration to authenticate with.',
+                'required': True,
+            },
+            {
+                'name': 'name',
+                'label': 'Repository Name',
+                'type': 'text',
+                'placeholder': 'e.g. getsentry/sentry',
+                'help': 'Enter your repository name, including the owner.',
+                'required': True,
+            },
+        ]
+
+    def validate_config(self, organization, config, actor=None):
+        """
+        ```
+        if config['foo'] and not config['bar']:
+            raise PluginError('You cannot configure foo with bar')
+        return config
+        ```
+        """
+        if config.get('name'):
+            client = self.get_client(config['integration_id'])
+            try:
+                repo = client.get_repo(config['name'])
+            except Exception as e:
+                self.raise_error(e)
+            else:
+                config['external_id'] = six.text_type(repo['uuid'])
+        return config
+
+    def get_webhook_secret(self, organization):
```

Comment:
Hmm... now that I look more closely at this one... does this still work as it should? Are we still using `OrganizationOptions`? @MaxBittker